tests/executables_src/testDataValidityPropagation.cc

```diff
@@ -0,0 +1,552 @@
+  //    auto pollInput2 = device1("dev.i2", typeid(int), 1, ChimeraTK::UpdateMode::poll);
+  //    auto trigger = m2("o1");
+  //    pollInput1[trigger] >> cs("dev.i1", typeid(int));
+  //    pollInput2[trigger] >> cs("dev.i2", typeid(int));
+  //  }
+};
+/**
+ *  tests the ExceptionDummyPollDecorator of the ExceptionDummyBackend, which provides a way for
+ *  forcing individual (poll-type) device outputs to DataValidity=faulty
+ */
+BOOST_AUTO_TEST_CASE(testDataValidity_exceptionDummy) {
+  TestApplication3 app;
+  ctk::TestFacility test;
+  //app.dumpConnections();
+
+  auto devI1 = test.getScalar<int>("/dev/i1");
+  auto devI2 = test.getScalar<int>("/dev/i2");
+  test.runApplication();
+  // it seems that the exceptionDummy is created only when app starts.
+  auto exceptionDummy =
+      boost::dynamic_pointer_cast<ctk::ExceptionDummy>(app.device1.getDeviceModule().device.getBackend());
+  assert(exceptionDummy);
+  exceptionDummy->setValidity("/dev/i1", ctk::DataValidity::faulty);
+
+  app.m2.o1 = 1;
+  app.m2.o1.write();
+  test.stepApplication();
+  devI1.read();
+  devI2.read();
+  BOOST_CHECK(devI1.dataValidity() == ctk::DataValidity::faulty);
+  BOOST_CHECK(devI2.dataValidity() == ctk::DataValidity::ok);
+}
+
+/*
+ * \anchor testDataValidity_1_1 \ref dataValidity_1_1 "1.1"
+ * In ApplicationCore each variable has a data validiy flag attached to it. DataValidity can be 'ok' or 'faulty'.
+ *
+ * Explicit test does not make sense since this is clear if this suite compiles,
+ * i.e. expressions testmod1.i1.dataValidity() and testmod1.o1.dataValidity();
+ */
+
+/*
+ * \anchor testDataValidity_1_2 \ref dataValidity_1_2 "1.2"
+ * This flag is automatically propagated: If any of the inputs of an ApplicationModule is faulty,
+ * the data validity of the module becomes faulty, which means all outputs of this module will automatically be
+ * flagged as faulty.
+ * Fan-outs might be special cases (see 2.4).
+ *
+ * See \ref testDataValidity_2_3_3
+ */
+
+/**
+ * \anchor testDataValidity_1_3 \ref dataValidity_1_3 "1.3"
+ * If a device is in error state, all variables which are read from it shall be marked as 'faulty'. This flag is then
+ * propagated through all the modules (via 1.2) so it shows up in the control system.
+ */
+BOOST_AUTO_TEST_CASE(testDataValidity_1_3) {
+  // set up an application with a faulty device
+  TestApplication1<TestModule1> app;
+  app.deviceError = true;
+  // testable mode cannot be used here, since it would wait on initial values (which are not provided)
+  ctk::TestFacility test(false);
+
+  auto i1 = test.getScalar<int>("/dev/i1");
+  test.runApplication();
+
+  // i1.read() would block here
+  i1.readLatest();
+
+  BOOST_CHECK(i1.dataValidity() == ctk::DataValidity::faulty);
+
+  // if Application does not shutdown, this could be an applicationcore bug that requires a workaround for this test
+  // redmine issue: #8550
+}
+
+/*
```

Comment:
Doxygen was merging it with the comment below it. Since there is no test here and merging comments anyway does not make sense, it is intended like written. Like this, all the spec items show up in the code in order.

Reply:
ah interesting, I didn't know Doxygen does that.